awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -374,6 +384,26 @@ def make_request(self, method, endpoint, *args, **kwargs):
         finally:
             self.url = self.url._replace(query=None)
 
+        if not self.version_checked:
+            # In PY2 we get back an HTTPResponse object but PY2 is returning an addinfourl
+            # First try to get the headers in PY3 format and then drop down to PY2.
+            try:
+                tower_type = response.getheader('X-API-Product-Name', None)
+                tower_version = response.getheader('X-API-Product-Version', None)
+            except Exception:
+                tower_type = response.info().getheader('X-API-Product-Name', None)
+                tower_version = response.info().getheader('X-API-Product-Version', None)
```

Comment:
My only remaining nit-pick is that this will not give a coherent error message when running against a version of AWX or Tower from before the header was introduced.

With the information in variables here, that could be turned into a specifically worded message, but it will involve more moving parts than I would like to put in. So I don't know quite where the balance should be struck, and we also have limited time to spend on the non-recommended use case.